pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -599,9 +574,24 @@ impl DeltaLayer {
             &self.path_or_conf,
             self.timeline_id,
             self.tenant_id,
-            &self.layer_name(),
+            &self.delta_layer_name(),
         )
     }
+
+    pub fn iter(&self) -> Result<LayerIter<'_>> {
+        let inner = self.load().context("load delta layer")?;
+        Ok(match DeltaValueIter::new(inner) {
+            Ok(iter) => Box::new(iter),
+            Err(err) => Box::new(std::iter::once(Err(err))),
```

Comment:
Is there a reason why we do not return plain error in this case? But an iterator containing one error value?